pkg/callout/callout.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package callout
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/absmach/supermq/pkg/errors"
+	svcerr "github.com/absmach/supermq/pkg/errors/service"
+)
+
+var errLimitExceeded = errors.New("limit exceeded")
+
+type Config struct {
+	CalloutURLs            []string      `env:"URLS"             envDefault:"" envSeparator:","`
+	CalloutMethod          string        `env:"METHOD"           envDefault:"POST"`
+	CalloutTLSVerification bool          `env:"TLS_VERIFICATION" envDefault:"true"`
+	CalloutTimeout         time.Duration `env:"TIMEOUT"          envDefault:"10s"`
+	CalloutCACert          string        `env:"CA_CERT"          envDefault:""`
+	CalloutCert            string        `env:"CERT"             envDefault:""`
+	CalloutKey             string        `env:"KEY"              envDefault:""`
+	CalloutPermissions     []string      `env:"INVOKE_PERMISSIONS" envDefault:"" envSeparator:","`
+}
+
+type callout struct {
+	httpClient        *http.Client
+	urls              []string
+	method            string
+	allowedPermission map[string]struct{}
+}
+
+// Callout send request to an external service.
+type Callout interface {
+	Callout(ctx context.Context, perm string, pl map[string]interface{}) error
+}
+
+// NewCallback creates a new instance of Callout.
+func NewCallout(ctls bool, certPath, keyPath, caPath string, timeout time.Duration, method string, urls []string, permissions []string) (Callout, error) {
+	httpClient, err := newCalloutClient(ctls, certPath, keyPath, caPath, timeout)
```

Comment:
Why don't we have something like this ?
```suggestion
func NewCallout(cfg Config) (Callout, error) {
```